JS/Lesson15/savas/script.js

```diff
@@ -0,0 +1,437 @@
+    if ((age >= 0) && (age <= 12)) {
+        console.log("You are a child.")
+    }
+
+    else if ((age >= 13) && (age <= 19)) {
+        console.log("You are teenager.")
+    }
+
+    else if ((age >= 20) && (age <= 64)) {
+        console.log("You are adult.")
+    }
+
+    else if (age >= 65) {
+        console.log("You are senior citizen.")
+    }
+
+    else {
+        console.log("Invalid value.")
+    }
+}
+
+checkAge(18);
+checkAge(24);
+checkAge(76);
+checkAge(50);
+checkAge(8);
+checkAge(-3);
+
+// 10. Write a program that takes age as a number and whether they are a student (yes or no). If the age is less than 18 and the user is a student, print "You are a student and not old enough to vote". If the age is between 18 and 65 (inclusive) and the user is not a student, print "You are eligible to vote". If the age is greater than 65, print "You are a senior citizen". If the user is a student but their age is greater than or equal to 18, print "You are old enough to vote".
+console.log("---------------------- Excersise 10: ---------------------")
+
+function appropriateForVote(age, student) {
+    if ((age < 18) && (student === "yes")) {
+        console.log("You are a student and not old enough to vote.")
+    }
+
+    else if ((age >= 18) && (age <= 65) && (student === "no")) {
+        console.log("You are eligible to vote.")
+    }
+
+    else if (age >= 65) {
+        console.log("You are a senior citizen.")
+    }
+
+    else if ((age >= 18) && (student === "yes")) {
+        console.log("You are old enough to vote.")
+    }
+    else {
+        console.log("Invalid value.")
+    }
+}
+
+appropriateForVote(18, "yes");
+appropriateForVote(24, "no");
+appropriateForVote(76, "no");
+appropriateForVote(50, "yes");
+appropriateForVote(8, "yes");
+appropriateForVote(-3, "no");
+
+// 11. Write a program that asks the user to enter their salary and their years of experience. If the salary is less than $30,000 and the years of experience are less than 5, print "Sorry, we cannot offer you the job at this time". If the salary is less than $50,000 and the years of experience are between 5 and 10 (inclusive), print "We can offer you the job at a lower salary". If the salary is greater than or equal to $50,000 or the years of experience are greater than or equal to 10, print "We can offer you the job at the requested salary".
+console.log("---------------------- Excersise 11: ---------------------")
+
+function offerSalary(salary, experience) {
+    if (salary < 30000 && experience < 5) {
+        console.log("Sorry, we cannot offer you the job at this time.")
+    }
+
+    else if (salary < 50000 && experience >= 5 && experience <= 10) {
+        console.log("We can offer you the job at a lower salary.")
+    }
+
+    else if (salary >= 50000 || experience >= 10) {
+        console.log("We can offer you the job at the requested salary.")
+    }
+
+    else {
+        console.log("Invalid value.")
+    }
+}
+
+offerSalary(20000, "3");
```

Comment:
Here's a problem! Inside a function, experience parameter is treated as a number. However, when calling a function, a string is passed to it (e.g. `"3"`). 
Since there's no type restriction in JS, it's good to always try to use variables in the expected  types when you can and add checks to see if the value is in the expected format.
E.g. I'd modify the function like this:
```
function offerSalary(salary, experience) {
  if (typeof salary !== 'number' || typeof experience !== 'number') { // should be the very first check
    console.log("Invalid input. Both salary and experience must be numbers.");
  } else if ...
....
```

Reply:
Ok, I see. Also, I left the else with invalid input, because negative numbers are also invalid input. Is it true method?